Improve CRM table

StefKors opened the pull request:
This PR does the following:

## Refactor flow actions and deprecate LinksProvider component

The LinksProvider component was designed to be used with a single SlideInFlowPanel and a single withLinks child component. Using it on a view wil multiple withLinks child components was not possible. This together with the introduction of optional link queries we have no reason to query for card links seperately. We can remove the LinksProvider logic and streamline our SlideInFlowPanel.

This PR also refactors the object stucture to be referenced on the channel target instead of just the card id. With this we can use a single SlideInFlowPanel component with multiple components controlling it's state.

***

**Please remember to write tests for your changes**. We aim to automatically
deploy `master` to production, and we can't safely do this without a solid
test suite!
<img width="1506" alt="Screen Shot 2020-09-22 at 6 22 04 PM" src="https://user-images.githubusercontent.com/11800807/93909842-89167880-fd00-11ea-9412-807906708565.png">

grahammcculloch reviewed (changes requested):
Not sure of the exact solution for the 'one SlideInFlowPanel but multiple cards' problem. That's the main thing I think we need to sort out on this branch. Otherwise just a bunch of minor React-y things!

StefKors committed:
Update view-all-opportunities to optionally query for owners

StefKors committed:
Update view-my-opportunities lens order

StefKors committed:
Remove slice from opportunity card

StefKors committed:
Refactor flow actions and deprecate LinksProvider component

The LinksProvider component was designed to be used with a single SlideInFlowPanel and a single withLinks child component. Using it on a view wil multiple withLinks child components was not possible. This together with the introduction of optional link queries we have no reason to query for card links seperately. We can remove the LinksProvider logic and streamline our SlideInFlowPanel.

This commit also refactors the object stucture to be referenced on the channel target instead of just the card id. With this we can use a single SlideInFlowPanel component with multiple components controlling it's state.

Change-type: major
Signed-off-by: Stef Kors <stef@balena.io>
